pathway-analysis/kegg_ortholog_mapping.Rmd

```diff
@@ -0,0 +1,252 @@
+If we want to use KEGG Pathways with another species without going through
+[KEGG Orthology](https://www.genome.jp/kegg/ko.html), we need to map to
+orthologs ourselves.
+
+We'll use the [`hcop` package](https://github.com/stephenturner/hcop) to do 
+this.
+If you're looking for a little bit more background information (like if you
+run into trouble installing `hcop`), check out the notebook in our
+[`ortholog-mapping`](https://github.com/AlexsLemonade/refinebio-examples/tree/master/ortholog-mapping) module.
+
+## Setup
+
+Package installation
+
+```{r}
+# need read.gmt functionality from qusage
+if (!("qusage" %in% installed.packages())) {
+  BiocManager::install("qusage", update = FALSE)
+}
+
+# we need devtools in order to install the hcop package we will use to
+# do the ortholog mapping
+if (!("devtools" %in% installed.packages())) {
+  install.packages("devtools")
+}
+
+# this installs a specific version of hcop
+# we pass the commit hash to the ref argument
+devtools::install_github("stephenturner/hcop", 
+                         ref = "0985fddc91a6ef2308f4800958dfd11c25fe6a98")
+```
+
+```{r}
+`%>%` <- dplyr::`%>%`
+```
+
+```{r}
+library(hcop)
+```
+
+## KEGG human pathways
+
+We need to download the the MSigDB v6.2 KEGG gene sets that use Entrez gene IDs
+and place them at the following path if we have not done so already:
+
+```
+gene-sets/c2.cp.kegg.v6.2.entrez.gmt
+```
+
+```{r}
+# the kegg gmt file should be located in the spot we mention above
+kegg_file <- file.path("gene-sets", "c2.cp.kegg.v6.2.entrez.gmt")
+# since we do not track this file in our repository, let's check to make sure
+# it exists where we expect it and download it if we don't find it
+if (!file.exists(kegg_file)) {
+  message(paste("KEGG GMT file is not found at", kegg_file, 
+                ", downloading now..."))
+  # need gene-sets directory
+  if (!dir.exists("gene-sets")) {
+    dir.create("gene-sets")
+  }
+  download.file("https://data.broadinstitute.org/gsea-msigdb/msigdb/release/6.2/c2.cp.kegg.v6.2.entrez.gmt", 
+                destfile = kegg_file)
+}
+```
+
+Read in the pathway file
+
+```{r}
+kegg_human_list <- qusage::read.gmt(kegg_file)
+```
+
+## Conversion from human Entrez ID to mouse symbol
+
+### Human to mouse mapping with `hcop`
+
+`hcop` is designed to work well with `dplyr`, so we'll get all the possible
+human Entrez IDs into a `data.frame`.
+
+```{r}
+entrez_in_pathways_df <- data.frame(
+  human_entrez = as.integer(unique(unlist(kegg_human_list)))
+)
+```
+
+Join to mouse orthologs by the human Entrez IDs.
+
+```{r}
+# Join to mouse orthologs
+mouse_ortholog_df <- entrez_in_pathways_df %>% 
+  dplyr::inner_join(mouse, by = "human_entrez")
+```
+
+### 1:many mapping
+
+For 1:many mappings, we'll pick the one with the most support.
+Here, we'll only consider the _number_ of sources without any regard for _what_
+those resources are.
+Essentially, we weigh all sources equally though they likely have more or less
```

Comment:
Is there a way that someone could use the permissiveness of the criteria for sorting or nah? Or what I'm guessing realistically the answer is probably: not easily here?

Reply:
I think you would have to make a choice about _weighting_ sources which I would probably encode in a column for each specific source. So the answer is yes you could do it, but it is beyond the scope of this example.